Algorithm.CSharp/CustomDataUnderlyingSymbolMappingRegressionAlgorithm.cs

```diff
@@ -0,0 +1,190 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using QuantConnect.Algorithm.Framework.Selection;
+using QuantConnect.Data;
+using QuantConnect.Data.Custom.SEC;
+using QuantConnect.Data.Fundamental;
+using QuantConnect.Data.UniverseSelection;
+using QuantConnect.Interfaces;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// Regression algorithm ensures that mapping is also applied to the underlying symbol(s) for custom data subscriptions
+    /// </summary>
+    /// <meta name="tag" content="using data" />
+    /// <meta name="tag" content="custom data" />
+    /// <meta name="tag" content="regression test" />
+    /// <meta name="tag" content="rename event" />
+    /// <meta name="tag" content="map" />
+    /// <meta name="tag" content="mapping" />
+    /// <meta name="tag" content="map files" />
+    public class CustomDataUnderlyingSymbolMappingRegressionAlgorithm : QCAlgorithm, IRegressionAlgorithmDefinition
+    {
+        private bool _initialSymbolChangedEvent;
+
+        // Equity to add custom data with as Symbol
+        private Symbol _equitySymbol;
+        private Symbol _badEquitySymbol;
+
+        // Custom data that was added with Symbol
+        private Symbol _customDataSymbol;
+        private Symbol _badCustomDataSymbol;
+
+        /// <summary>
+        /// Adds stocks GOOGL -&gt; GOOG and GOOG -&gt; GOOCV so that we can test if mapping occurs to the underlying symbol in the custom data subscription
+        /// as well as testing the behavior of adding custom data that can be mapped with a ticker
+        /// </summary>
+        public override void Initialize()
+        {
+            SetStartDate(2014, 3, 1);
+            SetEndDate(2014, 4, 9);
+            SetCash(100000);
+
+            UniverseSettings.Resolution = Resolution.Daily;
+
+            // Reset the symbol cache to test for ticker adding of custom data
+            SymbolCache.Clear();
```

Comment:
This shouldn't be required because the `AlgorithmRunner.RunLocalBacktest()` is already clearing the cache

Reply:
further -- I don't think we ever want to use `SymbolCache.Clear()` explicitly in our example/regression algorithms as it may lead to users actually using it, which would 99.999% of the time be incorrect.